### ISSUE TITLE
Please remove patch from composer.json

### ISSUE THREAD

**Issue by nccchris** [localgovdrupal/localgov_events#123]
"Date validation #3154944": "https://www.drupal.org/files/issues/2020-06-25/alpha-modal-form-end-date-validation.patch" [has been fixed](https://www.drupal.org/project/date_recur_modular/issues/3154944) and so can no longer be applied by composer : 
  
"https://www.drupal.org/files/issues/2020-06-25/alpha-modal-form-end-date-validation.patch (Date validation #3154944)
Could not apply patch! Skipping. The error was: Cannot apply patch https://www.drupal.org/files/issues/2020-06-25/alpha-modal-form-end-date-validation.patch"

**Comment by andybroomfield** [comment]
Are we sure the problem is fixed? It looks like its marked as a duplicate, so might need a rerolled patch from the duplicate issue.

@nccchris Does this prevent your deploys, or is it just warnings? Usually it should be possible to skip patches and the composer build continues, though that might be a setting in composer.json.

**Comment by nccchris** [comment]
It doesn't prevent deploys, it skips it with a red message. 
It's one of 12 such messages when I create a new install of microsites ( composer create-project --stability beta localgovdrupal/localgov_microsites_project TEST --no-install; cd TEST; composer install) which is a little off putting.

**Comment by andybroomfield** [comment]
I've opened a PR to try and remove the patch at #124.
However I'm not seeing any patch conflict, although the issue is marked as a duiplicate.

> Gathering patches for dependencies. This might take a minute.
>   - Installing drupal/date_recur_modular (3.1.0): Extracting archive
>   - Applying patches for drupal/date_recur_modular
>     https://www.drupal.org/files/issues/2020-06-25/alpha-modal-form-end-date-validation.patch (Date validation #3154944)

Thats not to say we shouldn't re-roll it as the issue has been marked as a duplicate and so no further devleopment is being made to that patch and it's likley to conflict with others.

Are you using any other patches @nccchris? 
Does anyone from @localgovdrupal/core-team who works on [Microsites](https://github.com/localgovdrupal/localgov_microsites) know if other patches are used for date recur?

**Comment by finnlewis** [comment]
Original issue: https://www.drupal.org/node/3154944 

Closed in favour of https://www.drupal.org/project/date_recur_modular/issues/3071666

Maybe we should test https://www.drupal.org/files/issues/2019-08-09/date_validation-3071666-09.patch

**Comment by finnlewis** [comment]
@nccchris 

when I run 

```
composer create-project --stability beta localgovdrupal/localgov_microsites_project TEST --no-install; cd TEST; composer install 
```

The patch does not fail for me.

I'm on :

PHP: 8.1.7
Composer: 2.5.1
Ubuntu : 23.04

What versions are you running on? 

Could you provide the output of your composer install?

**Comment by nccchris** [comment]
PHP 8.1.24
Composer 2.6.5
Redhat :  8.8 

Here's the warning I get about the patches. If you want the whole output just let me know.

- Applying patches for drupal/date_recur_modular
    https://www.drupal.org/files/issues/2020-06-25/alpha-modal-form-end-date-validation.patch (Date validation #3154944)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://www.drupal.org/files/issues/2020-06-25/alpha-modal-form-end-date-validation.patch

  - Applying patches for drupal/media_library_edit
    https://git.drupalcode.org/issue/media_library_edit-3315757/-/commit/9b590f1d93d8c0f750c50d90c422343ecf5e9e72.patch (Fix schema #3315757)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://git.drupalcode.org/issue/media_library_edit-3315757/-/commit/9b590f1d93d8c0f750c50d90c422343ecf5e9e72.patch

  - Applying patches for drupal/redirect
    https://www.drupal.org/files/issues/2022-09-01/3057250-53.patch (Validation issue on adding url redirect: https://www.drupal.org/project/redirect/issues/3057250)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://www.drupal.org/files/issues/2022-09-01/3057250-53.patch

  - Applying patches for drupal/viewsreference
    https://www.drupal.org/files/issues/2022-07-06/viewsreference_schema-2957529-13.patch (Fix schema #2957529)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://www.drupal.org/files/issues/2022-07-06/viewsreference_schema-2957529-13.patch

  - Applying patches for drupal/geolocation
    https://www.drupal.org/files/issues/2021-01-27/geolocation-google-maps-schema-update-3138668-5.patch (Fix schema #3138668)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://www.drupal.org/files/issues/2021-01-27/geolocation-google-maps-schema-update-3138668-5.patch

  - Applying patches for drupal/facets
    https://git.drupalcode.org/issue/facets-3311856/-/commit/765d5ef4228906c7f201e116763f3018a7867c96.patch (Don't render facet block if backend isn't available: https://www.drupal.org/project/facets/issues/3311856)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://git.drupalcode.org/issue/facets-3311856/-/commit/765d5ef4228906c7f201e116763f3018a7867c96.patch

  - Applying patches for drupal/groupmedia
    https://git.drupalcode.org/issue/groupmedia-3325129/-/commit/194f7bfa1d51f9a34a00ab4323f5fdf8c82ff48b.patch (Update addPage in groupmediato work with Group 3.x and suppress errors #3325129)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://git.drupalcode.org/issue/groupmedia-3325129/-/commit/194f7bfa1d51f9a34a00ab4323f5fdf8c82ff48b.patch

  - Applying patches for drupal/ginvite
    https://www.drupal.org/files/issues/2022-11-30/3324625-2-schema-fix-for-4.0.x.patch (View schema fix #3324625)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://www.drupal.org/files/issues/2022-11-30/3324625-2-schema-fix-for-4.0.x.patch

  - Applying patches for drupal/domain_path
    https://www.drupal.org/files/issues/2022-06-11/domain_path_pathauto-obey-current-domain_id-querying-for-reserved-aliases-3285213.patch (domain_path_pathauto creates duplicate aliases #3285213)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://www.drupal.org/files/issues/2022-06-11/domain_path_pathauto-obey-current-domain_id-querying-for-reserved-aliases-3285213.patch
    https://www.drupal.org/files/issues/2022-10-17/undefined-array-key-pathauto-3265497-2.patch (Warning: Undefined array key 'pathauto' #3315752)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://www.drupal.org/files/issues/2022-10-17/undefined-array-key-pathauto-3265497-2.patch
    https://raw.githubusercontent.com/localgovdrupal/localgov_microsites/995261d0909065e3124fba3fc0dce3e823aefa1d/patches/domain_path.146-url-aliases.patch (https://github.com/localgovdrupal/localgov_microsites/pull/175#issuecomment-1172879196)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://raw.githubusercontent.com/localgovdrupal/localgov_microsites/995261d0909065e3124fba3fc0dce3e823aefa1d/patches/domain_path.146-url-aliases.patch
    https://raw.githubusercontent.com/localgovdrupal/localgov_microsites/4cbdbe6ae3c3e95e7d2ed15d918c66805ad8e7f1/patches/localgov_microsites_group_326.domain_path_pathauto.unserializable.patch (https://github.com/localgovdrupal/localgov_microsites_group/issues/326)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://raw.githubusercontent.com/localgovdrupal/localgov_microsites/4cbdbe6ae3c3e95e7d2ed15d918c66805ad8e7f1/patches/localgov_microsites_group_326.domain_path_pathauto.unserializable.patch

**Comment by finnlewis** [comment]
Hey @nccchris 

Just checking in on this issue.

The patch mentioned still applies for me:   

> - Installing localgovdrupal/localgov_microsites (3.0.0-alpha1): Extracting archive
>   - Applying patches for drupal/date_recur_modular
>     https://www.drupal.org/files/issues/2020-06-25/alpha-modal-form-end-date-validation.patch (Date validation #3154944)
> 

Looking at your output, it looks like none of your patches were applying.

Is this still the case?

**Comment by nccchris** [comment]
Yes, I still get half a dozen errors per composer run on any of my LGD installs.

**Comment by andybroomfield** [comment]
Might be worth looking at [patches-ignore
](https://www.codeenigma.com/blog/composer-patches-patches-ignore#:~:text=Empowering%20Patch%20Management%20with%20patches%2Dignore) if it's hampering your own installs. and we can review if this patch is still needed.

Is anyone else experincing this issue?
